Legend refactor

chutko opened the pull request:
Improving network loading screen. 
- more flexible layout and better markup
- put #legend stylesheets into separate file
- write comments into the code
- use premade materialize components
- replacing network loader with materialize component
- use materialize page preloader with backdrop 


<!---
@huboard:{"order":484.0,"milestone_order":484,"custom_state":""}
-->

Commit:
style(legend): improve legend positioning and design:

- create separate #legend SASS component
- adjust network navbar position
- change few names to be more semantic
- facelift .loading-network positioning and design
- improve #legend backdrop position calculation
- make tidy in network z-indexes

Commit:
style(legend): legend footer design improvement

- blinking fade animatin on "Network Loaded" text

Commit:
style(legend,loading): different loading bar on network legend

- improved legend styles documentation
- improved stylesheets code

Commit:
style(legend): wrap all network legend parts under single container

Commit:
chore(legend): improve network-loaded state binding

Commit:
style(legend): use determinate loading bar

Commit:
style(loader): use materialize loader for page preloading

victor-nitu commented on app/templates/application.hbs:
```diff
@@ -3,16 +3,22 @@
 
+
+
+<div id="nav-wrapper">
```
Why do we need a wrapper around the nav?

chutko replied:
the nav positioning was acting unexpectedly. probably because it inherited many styles from materialize. this seemed like more comfortable solution at the time i was adjusting the network layout. but i agree its dirty, will have a look at it.

victor-nitu commented on app/templates/network/show.hbs:
```diff
@@ -1,84 +1,92 @@
-  </div>
+    {{! Shows when the network is loading and cannot be yet accessed }}
+    <div class="network-loader">
+      <span>Hold on, your network is loading..{{stabilizationPercent}}.</span>
```
You forgot to add a `%` sign here.

victor-nitu replied:
Also there are only two dots!

chutko replied:
oops, that was just for testing purposes. thanks for noticing.

victor-nitu commented on app/templates/network/show.hbs:
```diff
@@ -1,84 +1,92 @@
+    <div class="network-loader">
+      <span>Hold on, your network is loading..{{stabilizationPercent}}.</span>
+      <div class="progress">
+        <div class="determinate" style="width: {{concat stabilizationPercent "%"}}"></div>
```
I like this idea, so simple and effective!

victor-nitu reviewed (changes requested):
Apart from the code comments, there are a few bugs that need fixing before this can be merged. Here is how to reproduce them:

- on closing details view, sidebar goes full width
  - demo [here](https://elvis-ember-develop-pr-484.herokuapp.com/network/35b482f8-21cb-49fb-a9fe-42c401ed3f78/bidders)
  - click on a bidder, then close the details view
- sidebar toggle button is weirdly placed
- on clicking the sidebar toggle button, the view breaks

No more comments, everything looks awesome, just need to nail these things down and we're good!

Commit:
style(navbar): remove unnecessary navbar wrapper

-  remove forgotten debugging code from legen loading

Commit:
style(network): fix problems with sidebar closing

- adjust sidebar data table height
- adjust sidebar data table item text vertical positioning
- temporary disable network sidebar toggle button (always show sidebar)
- use different selector for styling network body

chutko commented:
okay, i have fixed victor's suggestions and had to fix some stuff around detail view selector. 

NOTE that I **disabled sidebar collapse/expand button** as it is not so much needed feature. i plan to refactor all network buttons in the future so dealing with this would be waste of time

victor-nitu reviewed (changes requested):
Query builder / cannot scroll the page, overflow is hidden.
![image](https://user-images.githubusercontent.com/1579997/54489656-44ae4f00-48b7-11e9-8a9e-0fa42782e2a8.png)

All the other issues are gone though.

Commit:
fix(query-builder): network-specific styles applying to query builder

chutko commented:
@nightsh fixed. its quite annoying that query builder `<body>` also receives `.network` class. i often forget about this

victor-nitu commented:
One more thing, this branch should merge into `develop`, not `master`.
